proposals/init.md

```diff
@@ -0,0 +1,582 @@
+whether or not this remains a binary breaking change.
+
+If we want to make the change of `init` to `set` a source and binary compatible
+change then it will force our hand on the modreq vs. attributes decision
+below because it will rule out modreqs as a soulution. If on the other hand
+this is seen as a non-interesting then this will make the modreq vs. attribute
+decision less impactful.
+
+**Resolution**
+This scenario is not seen as compelling by LDM.
+
+### Modreqs vs. attributes
+The emit strategy for `init` property accessors must choose between using 
+attributes or modreqs when emitting during metadata. These have different 
+trade offs that need to be considered.
+
+Annotating a property set accessor with a modreq declaration means CLI compliant
+compilers will ignore the accessor unless it understands the modreq. That means
+only compilers aware of `init` will read the member. Compilers unaware of 
+`init` will ignore the `set` member and hence will not accidentally treat the
+property as read / write. 
+
+The downside of modreq is `init` becomes a part of the binary signature of 
+the `set` accessor. Adding or removing `init` will break binary compatbility 
+of the application.
+
+Using attributes to annotate the `set` accessor means that only compilers which
+understand the attribute will know to limit access to it. A compiler unaware 
+of `init` will see it as a simple read / write property and allow access.
+
+This would seemingly mean this decision is a choice between extra safety at 
+the expense of binary compatibility. Digging in a bit the extra safety is not
+exactly what it seems. It will not protect against the following circumstances:
+
+1. Reflection over `public` members
+1. The use of `dynamic` 
+1. Compilers that don't recognize modreqs
+
+It should also be considered that, when we complete the IL verification rules 
+for .NET 5, `init` will be one of those rules. That means extra enforcement 
+will be gained from simply verifying compilers emitting verifiable IL.
+
+The primary languages for .NET (C#, F# and VB) will all be updated to 
+recognize these `init` accessors. Hence the only realistic scenario here is 
+when a C# 9 compiler emits `init` properties and they are seen by an older 
+toolset such as C# 8, VB 15, etc ... C# 8. That is the trade off to consider
+and weigh against binary compatibility.
+
+**Note**
+This discussion primarily applies to members only, not to fields. While `init`
+fields were rejected by LDM they are still interesting to consider for the 
+modreq vs. attribute discussion. The `init` feature for fields is a relaxation
+of the existing restriction of `readonly`. That means if we emit the fields as
+`readonly` + an attribute there is no risk of older compilers mis-using the 
+field because they would already recognize `readonly`. Hence using a modreq here
+doesn't add any extra protection.
+
+**Resolution**
+The feature will use a modreq to encode the property `init` setter. The
```

Comment:
Is this the appropriate place to comment about the decision?  I brought up [problems](https://github.com/dotnet/csharplang/issues/3323#issuecomment-616005297) with this approach in one of the issues for meeting notes which I think make `modreq` an unattractive option for encoding `init` setters.  I imagine the details of which will be a subject for a future language design meeting.

Reply:
Let's use a separate issue to discuss this aspect. Want to keep this PR focused on recording the design + the current decision. I will ping the developer who's working on this feature @jcouv to make sure he sees that comment so we can make sure we get to the bottom of it.

Reply:
I've submitted a [separate issue](https://github.com/dotnet/csharplang/issues/3376) to discuss this problem.  I also think I've found a solution which I included in that issue and doesn't deviate a great deal from this design.